local_planner/src/nodes/planner_functions.cpp

```diff
@@ -496,10 +504,9 @@ bool getDirectionFromTree(
           (1.f - l_frac) * toEigen(path_node_positions[wp_idx - 1]) +
           l_frac * toEigen(path_node_positions[wp_idx]);
 
-      int wp_e = floor(elevationAnglefromCartesian(mean_point, position));
-      int wp_z = floor(azimuthAnglefromCartesian(mean_point, position));
+      PolarPoint wp_pol = CartesianToPolar(mean_point, position);
 
-      p = Eigen::Vector3f(wp_e, wp_z, 0.f);
+      p = Eigen::Vector3f(wp_pol.e, wp_pol.z, 0.f);
```

Comment:
I would change the function to output a polar point. Here you mix up the types again. It would be nicer if wp_pol would be the output, such that also seen from the outside, you know that you are getting the waypoint in polar coordinates, not in cartesian ones.